core/src/main/java/org/elasticsearch/plugins/PluginsService.java

```diff
@@ -206,8 +207,13 @@ public PluginsService(Settings settings, Path modulesDirectory, Path pluginsDire
                 }
                 Class moduleClass = method.getParameterTypes()[0];
                 if (!Module.class.isAssignableFrom(moduleClass)) {
-                    logger.warn("Plugin: {} implementing onModule by the type is not of Module type {}", pluginEntry.v1().getName(), moduleClass);
-                    continue;
+                    String extraMessage = "";
```

Comment:
Not needed, nah. I wanted to keep the friendliness we had with the final deprecated method that I removed. I guess I could just leave the deprecated method and skip it silently? I could also just remove it and rely on the developer to read the breaking changes docs, but I figure being friendly doesn't really hurt.